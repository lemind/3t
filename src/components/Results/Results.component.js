import React from 'react';
import './Results.less';
import { CONFIG } from './../../config';

export class ResultsComponent extends React.Component {
  constructor(props) {
    super();

    const { started, winner, playerSign } = {...props.process}

    this.state = {
      started: props.process.started,
      winner: props.process.winner
    };
  }

  getCurrentStatus() {
    const state = this.state;

    if (!state.started) {
      return 'Ready to start'
    } else {
      if (state.winner) {
        if (state.winner === 'draw') {
          return 'Draw'
        } else {
          return state.winner === CONFIG.PLAYER_SIGN
            ? 'You win!'
            : 'You lose'
        }
      } else {
        return 'In progress'
      }
    }
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    return {
      started: nextProps.process.started,
      winner: nextProps.process.winner
    }
  }

  render() {
    return (
      <div className="results">
        <span className="results__label">Result:</span>
        <span className="results__status">{ this.getCurrentStatus() }</span>
      </div>
    )
  };
}
