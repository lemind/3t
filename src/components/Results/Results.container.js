import { connect } from 'react-redux';

import { ResultsComponent } from './Results.component';

export const ResultsContainer = connect(
  function mapStateToProps(state) {
    return {
      process: state.process
    };
  },
  function mapDispatchToProps(dispatch) {
    return {}
  }
)(ResultsComponent);