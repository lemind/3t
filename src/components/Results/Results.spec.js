import React from 'react';
import { shallow } from 'enzyme';
import { ResultsComponent as Results } from './Results.component';

describe('Results', () => {

  beforeEach(() => {
  })

  test('should render Results', () => {
    expect(shallow(<Results process={ {} } />)).toMatchSnapshot();
  })
})
