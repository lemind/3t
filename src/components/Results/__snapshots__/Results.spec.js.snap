// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render Results 1`] = `
<div
  className="results"
>
  <span
    className="results__label"
  >
    Result:
  </span>
  <span>
    Ready to start
  </span>
</div>
`;
